torch/_subclasses/meta_utils.py

```diff
@@ -246,22 +246,174 @@ def sym_sizes_strides_storage_offset(
                     return shape_env.create_symbolic_sizes_strides_storage_offset(
                         t,
                         src,
-                        # Assume that the set of dims that are dynamic are the same between
-                        # the wrapper tensor and any inner tensors.
-                        # We can revisit this if this assumption does not hold
-                        # for any important subclasses later.
-                        policy=policy,
+                        policy=inner_policy,
                     )
             else:
-                assert policy is None
+                assert inner_policy is None
             return (t.size(), t.stride(), t.storage_offset())
 
+        def empty_create(
+            inner_t,
+            inner_src,
+            inner_policy=policy,
+        ):
+            (
+                inner_sizes,
+                inner_strides,
+                inner_storage_offset,
+            ) = sym_sizes_strides_storage_offset(
+                inner_t,
+                inner_src,
+                inner_policy=inner_policy,
+            )
+            return torch.empty_strided(
+                inner_sizes,
+                inner_strides,
+                dtype=inner_t.dtype,
+                device="meta",
+            )
+
         # see expired-storages
         self.check_expired_count += 1
         if self.check_expired_count >= self.check_expired_frequency:
             self.check_for_expired_weak_storages()
             self.check_expired_count = 0
 
+        # Meta-ifies a nested tensor, maintaining proper view relationships.
+        # Assumes any base has already been meta-ified.
```

Comment:
The rest of the code here tends to recursively metafy base as part of the metafication process, not sure if this is breaking convention or you changed the rest of it